### DISCUSSION TITLE
Error message after building MOOSE

### DISCUSSION THREAD

**Discussion by Emmanuel-324** [idaholab/moose#27294] [Q&A Getting Started]
So I encountered an error while updating my MOOSE so I tried to start everything from scratch. But during the build of MOOSE I encountered the error below. I git clean -xfd and rebuilt WASP but still did not work. Below is the error message.

n file included from /home/emmanuel324/projects/moose/framework/build/unity_src/base_Unity.C:21:
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C: In member function 'bool MooseServer::addParametersToList(wasp::DataArray&, const InputParameters&, const std::set<std::__cxx11::basic_string<char> >&, int, int, int, int, const std::string&)':
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:529:9: error: 'client_snippet_support' was not declared in this scope
  529 |     if (client_snippet_support && !default_value.empty())
      |         ^~~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:531:32: error: 'm_text_format_snippet' is not a member of 'wasp::lsp'
  531 |       text_format = wasp::lsp::m_text_format_snippet;
      |                                ^~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:536:32: error: 'm_text_format_plaintext' is not a member of 'wasp::lsp'
  536 |       text_format = wasp::lsp::m_text_format_plaintext;
      |                                ^~~~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:544:45: error: too many arguments to function 'bool wasp::lsp::buildCompletionObject(wasp::DataObject&, std::ostream&, const std::string&, int, int, int, int, const std::string&, int, const std::string&, const std::string&, bool, bool)'
  544 |     pass &= wasp::lsp::buildCompletionObject(*item,
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
  545 |                                              errors,
      |                                              ~~~~~~~
  546 |                                              param_name,
      |                                              ~~~~~~~~~~~
  547 |                                              replace_line_beg,
      |                                              ~~~~~~~~~~~~~~~~~
  548 |                                              replace_char_beg,
      |                                              ~~~~~~~~~~~~~~~~~
  549 |                                              replace_line_end,
      |                                              ~~~~~~~~~~~~~~~~~
  550 |                                              replace_char_end,
      |                                              ~~~~~~~~~~~~~~~~~
  551 |                                              insert_text,
      |                                              ~~~~~~~~~~~~
  552 |                                              complete_kind,
      |                                              ~~~~~~~~~~~~~~
  553 |                                              "",
      |                                              ~~~
  554 |                                              doc_string,
      |                                              ~~~~~~~~~~~
  555 |                                              false,
      |                                              ~~~~~~
  556 |                                              false,
      |                                              ~~~~~~
  557 |                                              text_format);
      |                                              ~~~~~~~~~~~~
In file included from /home/emmanuel324/projects/moose/framework/build/header_symlinks/MooseServer.h:13,
                 from /home/emmanuel324/projects/moose/framework/src/base/MooseApp.C:53,
                 from /home/emmanuel324/projects/moose/framework/build/unity_src/base_Unity.C:13:
/home/emmanuel324/projects/moose/framework/contrib/wasp/install/include/wasplsp/LSP.h:604:6: note: declared here
  604 | bool buildCompletionObject( DataObject        & object          ,
      |      ^~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C: In member function 'bool MooseServer::addSubblocksToList(wasp::DataArray&, const std::string&, int, int, int, int, const std::string&, bool)':
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:618:25: error: 'client_snippet_support' was not declared in this scope
  618 |       int text_format = client_snippet_support ? wasp::lsp::m_text_format_snippet
      |                         ^~~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:618:61: error: 'm_text_format_snippet' is not a member of 'wasp::lsp'
  618 |       int text_format = client_snippet_support ? wasp::lsp::m_text_format_snippet
      |                                                             ^~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:619:61: error: 'm_text_format_plaintext' is not a member of 'wasp::lsp'
  619 |                                                : wasp::lsp::m_text_format_plaintext;
      |                                                             ^~~~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:641:47: error: too many arguments to function 'bool wasp::lsp::buildCompletionObject(wasp::DataObject&, std::ostream&, const std::string&, int, int, int, int, const std::string&, int, const std::string&, const std::string&, bool, bool)'
  641 |       pass &= wasp::lsp::buildCompletionObject(*item,
      |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
  642 |                                                errors,
      |                                                ~~~~~~~
  643 |                                                subblock_name,
      |                                                ~~~~~~~~~~~~~~
  644 |                                                replace_line_beg,
      |                                                ~~~~~~~~~~~~~~~~~
  645 |                                                replace_char_beg,
      |                                                ~~~~~~~~~~~~~~~~~
  646 |                                                replace_line_end,
      |                                                ~~~~~~~~~~~~~~~~~
  647 |                                                replace_char_end,
      |                                                ~~~~~~~~~~~~~~~~~
  648 |                                                insert_text,
      |                                                ~~~~~~~~~~~~
  649 |                                                complete_kind,
      |                                                ~~~~~~~~~~~~~~
  650 |                                                "",
      |                                                ~~~
  651 |                                                doc_string,
      |                                                ~~~~~~~~~~~
  652 |                                                false,
      |                                                ~~~~~~
  653 |                                                false,
      |                                                ~~~~~~
  654 |                                                text_format);
      |                                                ~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/contrib/wasp/install/include/wasplsp/LSP.h:604:6: note: declared here
  604 | bool buildCompletionObject( DataObject        & object          ,
      |      ^~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C: In member function 'bool MooseServer::addValuesToList(wasp::DataArray&, const InputParameters&, const 
std::set<std::__cxx11::basic_string<char> >&, const std::string&, const std::set<std::__cxx11::basic_string<char> >&, int, int, int, int)':
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:804:57: error: 'm_text_format_plaintext' is not a member of 'wasp::lsp'
  804 |                                              wasp::lsp::m_text_format_plaintext);
      |                                                         ^~~~~~~~~~~~~~~~~~~~~~~
Compiling C++ (in opt mode) /home/emmanuel324/projects/moose/framework/build/unity_src/geomsearch_Unity.C...
Compiling C++ (in opt mode) /home/emmanuel324/projects/moose/framework/build/unity_src/postprocessors_Unity.C...
make: *** [/home/emmanuel324/projects/moose/framework/build/unity_src/base_Unity.x86_64-pc-linux-gnu.opt.lo] Error 1
make: *** Waiting for unfinished jobs....
(python-stack) [emmanuel324@tinkercliffs1 test]$

**Comment by GiudGiud** [comment]
Hello

this is because your version of wasp in not in sync with your version of moose.

If you used the conda installation path, then you need to ‘conda update moose-dev’
If you are building everything from source, then the update wasp script in moose/scripts should have fixed this

Guillaume

**Comment by GiudGiud** [comment]
Ok seems @brandonlangley is helping you on the other thread. I am closing this one, please avoid double posting in the future.
